From 3a283522b661feb9309ad52363677593316602d2 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 14 Dec 2016 13:02:24 +0800
Subject: [PATCH 1358/5242] MLK-13608: mfd: mxc-hdmi-core: fix the dependency
 of alsa sound

commit  38fb0e7a0e0b86f139a3d66deb1975eaf15101ab from
https://source.codeaurora.org/external/imx/linux-imx.git

When remove sound card build config, there will be error

drivers/built-in.o: In function `mxc_hdmi_register_audio':
vf610_adc.c:(.text+0xd64f8): undefined reference to `_snd_pcm_stream_lock_irqsave'
vf610_adc.c:(.text+0xd6518): undefined reference to `snd_pcm_stream_unlock_irqrestore'
drivers/built-in.o: In function `mxc_hdmi_abort_stream':
vf610_adc.c:(.text+0xd65a4): undefined reference to `_snd_pcm_stream_lock_irqsave'
vf610_adc.c:(.text+0xd65d0): undefined reference to `snd_pcm_stream_unlock_irqrestore'
make: *** [vmlinux] Error 1

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mfd/mxc-hdmi-core.c |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/mxc-hdmi-core.c b/drivers/mfd/mxc-hdmi-core.c
index a670406..6435b3d 100644
--- a/drivers/mfd/mxc-hdmi-core.c
+++ b/drivers/mfd/mxc-hdmi-core.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2011-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2011-2016 Freescale Semiconductor, Inc.
 
  * The code contained herein is licensed under the GNU General Public
  * License. You may obtain a copy of the GNU General Public License
@@ -100,6 +100,7 @@ unsigned int hdmi_set_blank_state(unsigned int state)
 }
 EXPORT_SYMBOL(hdmi_set_blank_state);
 
+#ifdef CONFIG_SND_SOC_IMX_HDMI_DMA
 static void hdmi_audio_abort_stream(struct snd_pcm_substream *substream)
 {
 	unsigned long flags;
@@ -130,6 +131,13 @@ int mxc_hdmi_abort_stream(void)
 	return 0;
 }
 EXPORT_SYMBOL(mxc_hdmi_abort_stream);
+#else
+int mxc_hdmi_abort_stream(void)
+{
+	return 0;
+}
+EXPORT_SYMBOL(mxc_hdmi_abort_stream);
+#endif
 
 int check_hdmi_state(void)
 {
@@ -148,6 +156,7 @@ int check_hdmi_state(void)
 }
 EXPORT_SYMBOL(check_hdmi_state);
 
+#ifdef CONFIG_SND_SOC_IMX_HDMI_DMA
 int mxc_hdmi_register_audio(struct snd_pcm_substream *substream)
 {
 	unsigned long flags, flags1;
@@ -175,6 +184,7 @@ int mxc_hdmi_register_audio(struct snd_pcm_substream *substream)
 	return ret;
 }
 EXPORT_SYMBOL(mxc_hdmi_register_audio);
+#endif
 
 void mxc_hdmi_unregister_audio(struct snd_pcm_substream *substream)
 {
-- 
1.7.9.5

